### ISSUE TITLE
let and const variables have different rules for annotations but shouldn't

### ISSUE THREAD

**Issue by mina86** [google/closure-compiler#749]
```
$ cat /tmp/decl.js 
/** @param {number} n */
var var_arrow = n => n + 1;

/** @param {number} n */
let let_arrow = n => n + 1;

/** @param {number} n */
const const_arrow = n => n + 1;

/** @param {number} n */
var var_func = function(n) { return n + 1; };

/** @param {number} n */
let let_func = function(n) { return n + 1; };

/** @param {number} n */
const const_func = function(n) { return n + 1; };
$ java -jar compiler.jar --js_output_file=/dev/null \
    --language_in=ECMASCRIPT6_STRICT --language_out=ES5 \
    /tmp/decl.js
/tmp/decl.js:5: WARNING - Misplaced function annotation.
let let_arrow = n => n + 1;
^

/tmp/decl.js:8: WARNING - Misplaced function annotation.
const const_arrow = n => n + 1;
^

/tmp/decl.js:14: WARNING - Misplaced function annotation.
let let_func = function(n) { return n + 1; };
^

/tmp/decl.js:17: WARNING - Misplaced function annotation.
const const_func = function(n) { return n + 1; };
^

0 error(s), 4 warning(s)
```

Expected result: no warnings and the type annotations applied to the function bound to the variable regardless of which of var, let or const has been used.
